ingalls/Hecate
# Session Invalidation

### Context

At the moment, if a session cookie is set and invalid, it will block all calls to hecate, not redirecting the user to auth. This forces the user to go and manually delete the cookie before they are able to reauth.

This implements automatic session cookie clearing on any invalid auth attempt that uses a session cookie.

@lizziegooding or @mattciferri can I get a review and have you test/take it out?

### Test Instructions

- Run any test that creates the `ingalls:yeaheh` user/pass
- Start server locally with 
```
cargo run -- --auth test/fixtures/auth.default.json
```
- navigate to `localhost:8000/admin`
- Should be redireted to login page, login with `ingalls:yeaheh` user:pass
- Run
```
psql -U postgres hecate
```
```
DELETE FROM users_tokens
```

- Refresh the page, you should be deauthenticated and returned to the login page, instead of seeing the white page you may be used to

cc/ @ingalls
